tpp-app/src/home/pages/CycleHistoryScreen.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect } from 'react';
+import {SafeAreaView, Text, StyleSheet, View, TouchableOpacity, ImageBackground} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import background from '../../../ios/tppapp/Images.xcassets/SplashScreenBackground.imageset/watercolor-background.png';
+import {ExpandedHistoryCard} from '../components/CycleHistory';
+import CycleService from '../../services/cycle/CycleService';
+import {GETStoredYears} from '../../services/utils/helpers';
+
+function Header({navigation}){
+    return(
+    <View style={ styles.headerContainer}>
+        <View
+        style={[styles.nonCenterComponent]}
+        >
+            <TouchableOpacity 
+                onPress={() =>navigation.goBack()}
+            >
+                <Icon name="keyboard-arrow-left" size={36} color={"#5A9F93"}/>
+            </TouchableOpacity>
+        </View>
+        <Text style={styles.headerText}> Period History </Text>
+        <View style={styles.nonCenterComponent}><Text></Text></View>
+    </View>
+    );
+}
+
+function YearButton({year, selectedYear, setSelectedYear}){
+
+    let backgroundColor = year == selectedYear ? "#B31F20" : "#FFFFFF";
+    let textColor = year == selectedYear ? "#FFFFFF" : "#C4C4C4";
+    let border = year == selectedYear ? null : styles.buttonBorder;
+    return (
+        <TouchableOpacity 
+            onPress={() => setSelectedYear(year)}
+            style= {[styles.button, border, {backgroundColor: backgroundColor}]}
+        >
+            <Text style={{color: textColor}}>{year}</Text>
+        </TouchableOpacity>
+    )
+}
+
+export default function CycleHistoryScreen({navigation}){
+    let [currentIntervals, setCurrentIntervals] = useState([]);
+    let [selectedYear, setSelectedYear] = useState(new Date().getFullYear());
+    let [storedYears, setStoredYears] = useState([]);
+    let [onPeriod, setOnPeriod] = useState(false);
+    
+    useEffect(() => {
+        GETStoredYears().then(
+            years => {
+                setStoredYears(years);
+            }
+        )
+        CycleService.GETPeriodDay().then(days => {
+            setOnPeriod(days != 0 );
+        })
```

Comment:
You should either put `catch` blocks for these to avoid getting an unresolved Promise rejection warning. Or, you also might want to make these `await`. In which case, you'll have to first declare an async function and then call it like so:
```javascript
  useEffect(() => {
      async function fetchData() {
        let years = await getStoredYears();
        setStoredYears(years);
       ....
      }
      fetchData()
    }
  }, [])
```

Reply:
Same applies to the `useEffect` below